selfservice/utilities/reset.py

```diff
@@ -107,15 +108,18 @@ def passwd_reset(username, password):
     # Create LDAP admin session to perform initial reset.
     dn = "uid={},cn=users,cn=accounts,dc=csh,dc=rit,dc=edu".format(username)
 
-    l = ldap.initialize("ldaps://stone.csh.rit.edu")
+    # Find FreeIPA server
+    ldap_srvs = srvlookup.lookup("ldap", "tcp", "csh.rit.edu")
+    ldap_uri = ldap_srvs[0].host
+    l = ldap.initialize("ldaps://"+ldap_uri)
     l.simple_bind_s(app.config["LDAP_BIND_DN"], app.config["LDAP_BIND_PW"])
     l.modify_s(dn, [(ldap.MOD_REPLACE, "userPassword", [password.encode()])])
     l.modify_s(dn, [(ldap.MOD_REPLACE, "nsaccountlock", ["false".encode()])])
 
     # FreeIPA automatically expires the password set through the previous
     # method, so we need to use their password change API to get past that.
     requests.post(
-        "https://stone.csh.rit.edu/ipa/session/change_password",
+        "https://"+ldap_uri+"/ipa/session/change_password",
```

Comment:
You might consider using f strings. I tend to find them more readable than concatenation, but in a case like this it’s entirely your preference.

Reply:
Actually, you definitely should use format(). If nothing else, it is consistent with the rest of the application.

Reply:
Switched it to format in all locations